packages/stateful/actions/actions/ValidatorActions.tsx

```diff
@@ -0,0 +1,158 @@
+
+import { PickEmoji } from '@dao-dao/stateless'
+import {
+  ActionComponent,
+  ActionMaker,
+  ActionOptionsContextType,
+  CoreActionKey,
+  UseDecodedCosmosMsg,
+  UseDefaults,
+  UseTransformToCosmos,
+} from '@dao-dao/types/actions'
+import { NATIVE_DENOM, makeStargateMessage } from '@dao-dao/utils'
+
+import { ValidatorActionsComponent as StatelessValidatorActionsComponent } from '../components'
+
+export enum ValidatorActionType {
+  CreateValidator = '/cosmos.staking.v1beta1.MsgCreateValidator',
+  EditValidator = '/cosmos.staking.v1beta1.MsgEditValidator',
+  UnjailValidator = '/cosmos.slashing.v1beta1.MsgUnjail',
+  WithdrawValidatorCommission = '/cosmos.distribution.v1beta1.MsgWithdrawValidatorCommission',
+}
+
+interface ValidatorActionsData {
+  createMsg: string
+  editMsg: string
+  unjailMsg: MsgUnjail
+  validatorActionType: ValidatorActionType
+  withdrawCommissionMsg: MsgWithdrawValidatorCommission
+}
+
+const useDefaults: UseDefaults<ValidatorActionsData> = () => {
+  return {
+    validatorActionType: ValidatorActionType.WithdrawValidatorCommission,
+    createMsg: `{
+   "description": {
+     "moniker": "<validator name>",
+     "identity": "<optional identity signature (ex. UPort or Keybase)>",
+     "website": "<your validator website>",
+     "securityContact": "<optional security contact email>",
+     "details": "<description of your validator>"
+   },
+   "commission": {
+     "rate": "50000000000000000",
+     "maxRate": "200000000000000000",
+     "maxChangeRate": "100000000000000000"
+   },
+   "minSelfDelegation": "1",
+   "delegatorAddress": "<your DAO address>",
+   "validatorAddress": "<your DAO validator address>",
+   "pubkey": {
+     "typeUrl": "/cosmos.crypto.ed25519.PubKey",
+     "value": {
+       "@type":"/cosmos.crypto.ed25519.PubKey",
+       "key":"<the public key of your node (junod tendermint show-validator)>"}
+   },
+   "value": {
+     "denom": "${NATIVE_DENOM}",
+     "amount": "1000000"
+   }
+ }`,
+    editMsg: `{
+   "description": {
+     "moniker": "<validator name>",
+     "identity": "<optional identity signature (ex. UPort or Keybase)>",
+     "website": "<your validator website>",
+     "securityContact": "<optional security contact email>",
+     "details": "<description of your validator>"
+   },
+   "commissionRate": "50000000000000000",
+   "minSelfDelegation": "1",
+   "validatorAddress": "<your validator address>"
+ }`,
+    unjailMsg: {
+      validatorAddr: '',
+    },
+    withdrawCommissionMsg: {
+      validatorAddress: '',
+    },
+  }
+}
+
+const Component: ActionComponent = (props) => {
+  return <StatelessValidatorActionsComponent {...props} options={{}} />
+}
+
+const useDecodedCosmosMsg: UseDecodedCosmosMsg<ValidatorActionsData> = (
+  msg: Record<string, any>
+) =>
+  useMemo(
+    () =>
+      msg
+        ? {
+            match: true,
+            data: msg as ValidatorActionsData,
+          }
+        : { match: false },
+    [msg]
+  )
+
```

Comment:
this will make it think all messages are validator actions messages. the goal of this hook is to recognize from a CosmosMsg whether or not it is of this action type. right now, all actions will try to render as validator actions 🤪